satellite/satellitedb/orders.go

```diff
@@ -236,6 +236,12 @@ func (db *ordersDB) ProcessOrders(ctx context.Context, requests []*orders.Proces
 	if len(requests) == 0 {
 		return nil, err
 	}
+	sort.Slice(requests, func(i, k int) bool {
```

Comment:
is this logic tested somewhere already?

Reply:
Not really, but @ethanadams has a reproducer.

Reply:
@navillasa are you referring to testing the sort logic, or testing the fact that this change fixes the deadlock issue?

Reply:
I think I meant testing the sort logic, but maybe testing that it fixes the deadlock is more important.

Reply:
Apparently my test was not comprehensive enough.  I was uploading a lot of small files and then restarting the sim.  This did reproduce the issue before the sorting.  I received some help from @littleskunk and have updated my script to this:
```
$ for (( c=10081; c<=10100; c++ )) do uplink --config-dir "/Users/ethan/Library/Application Support/Storj/masterpg" cp ~/Downloads/< 70M file > sj://test3/$c; uplink --config-dir "/Users/ethan/Library/Application Support/Storj/masterpg" cp sj://test3/$c /dev/null; done
```
I can now reproduce.